tests/KoPrivateGPT/retrieval/test_base_retrieval.py

```diff
@@ -47,13 +52,128 @@ def validate_passages(retrieved_passage: List[Passage], top_k: int):
+    if not os.path.exists(os.path.dirname(bm25_path)):
+        os.makedirs(os.path.dirname(bm25_path))
+    bm25_retrieval = BM25Retrieval(save_path=bm25_path)
+    ready_pickle_db(pickle_path)
+    yield bm25_retrieval
+    # teardown
+    if os.path.exists(bm25_path):
+        os.remove(bm25_path)
+    if os.path.exists(pickle_path):
+        os.remove(pickle_path)
+
+
+def test_duplicate_check(just_bm25_retrieval):
+    assert just_bm25_retrieval.duplicate_check(TEST_DB_ORIGIN) == TEST_DB_ORIGIN_RESULT
+
+
+def test_is_created(just_bm25_retrieval):
+    """
+    For 'is_created'
+    1. db_instance_list가 비어있을 때, list에 db instance가 잘 들어가는가? 해당 db를 잘 return하는가?
+    2. db_instance_list가 비어있지 않고, db_origin이 이미 존재하는 경우, 해당 db를 잘 return하는가?
+    3. db_instance_list가 비어있지 않고, db_origin이 존재하지 않는 경우, list에 db instance가 잘 들어가는가? 해당 db를 잘 return하는가?
+    """
+    # 0. reset db_instance_list for test
+    just_bm25_retrieval.db_instance_list = []
+    # 1. If 'db_instance_list' is empty
+    first_instance = just_bm25_retrieval.is_created(db_type=TEST_DB_ORIGIN[0]['db_type'],
+                                                    db_path=TEST_DB_ORIGIN[0]['db_path'])
+    assert first_instance == just_bm25_retrieval.db_instance_list[0]
+    # 2. db_instance_list is not empty, db_origin already exists
+    second_instance = just_bm25_retrieval.is_created(db_type=TEST_DB_ORIGIN[0]['db_type'],
+                                                     db_path=TEST_DB_ORIGIN[0]['db_path'])
+    assert second_instance == just_bm25_retrieval.db_instance_list[0]
+    # 3. db_instance_list is not empty, db_origin does not already exist
+    third_instance = just_bm25_retrieval.is_created(db_type=TEST_DB_ORIGIN[1]['db_type'],
+                                                    db_path=TEST_DB_ORIGIN[1]['db_path'])
+    assert third_instance == just_bm25_retrieval.db_instance_list[1]
+
+
+TEST_DB_ORIGIN_RESULT_2 = {(('db_type', 'mongo_db'),
+                            ('db_path', (('mongo_url', f'{os.getenv("MONGO_URL")}'),
+                                         ('db_name', f'{os.getenv("MONGO_DB_NAME")}'),
+                                         ('collection_name', 'test_retrieval'))))
+                           : [0],
+                           (('db_type', 'mongo_db'),
+                            ('db_path', (('mongo_url', f'{os.getenv("MONGO_URL")}'),
+                                         ('db_name', f'{os.getenv("MONGO_DB_NAME")}'),
+                                         ('collection_name', 'test_retrieval_2'))))
+                           : [1]
+                           }
+
+TEST_PASSAGES_2 = [TEST_PASSAGES[0]]
+TEST_PASSAGES_3 = [TEST_PASSAGES[1]]
+TEST_RESULT_PASSAGES = [TEST_PASSAGES[0], TEST_PASSAGES[1]]
+
+TEST_IDS = [TEST_PASSAGES[0].id, TEST_PASSAGES[1].id]
+
 
-TEST_IDS = [passage.id for passage in TEST_PASSAGES]
+def test_fetch_each_db(just_bm25_retrieval):
+    # Create db_instance
+    mongo_db = MongoDB(
```

Comment:
make mongodb at pytest.fixture and teardown (delete collection) after usage. You can get multiple fixtures as test_fetch_each_db function parameter.